packages/cli-kit/src/public/node/progress-indicators.ts

```diff
@@ -0,0 +1,68 @@
+import colors from './colors.js'
+import {twoThirdsOfTerminalWidth} from './terminal-layout.js'
+import gradient from 'gradient-string'
+
+const ANIMATION_HUE_SPEED = 5
+const GRADIENT_HUE_SPREAD = 60
+const MAX_DOTS = 3
+
+/**
+ * Creates a colored progress bar showing completion percentage.
+ *
+ * @param percentage - The completion percentage (0-100).
+ * @returns The colored progress bar string.
+ */
+export function createColoredProgressBar(percentage: number): string {
+  const width = twoThirdsOfTerminalWidth()
```

Comment:
I'd rather pass down `width` as function argument instead so progress bar could be more customizable.

Reply:
good idea!

Reply:
Done, thanks :)